### PR TITLE
fix: use default IAM endpoint when fetching IMDSv2 token

### PR DESCRIPTION
**Pull request by iler-dev**
If a custom IAM endpoint is not specified, the IAM credential provider did not use the default endpoint to fetch the IMDSv2 token. This changes sets the default earlier so it is used for fetching the token.

### CODE REVIEW

File: pkg/credentials/iam_aws.go [hunk 1]
```diff
@@ -181,10 +181,6 @@ type ec2RoleCredRespBody struct {
 // be sent to fetch the rolling access credentials.
 // http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html
 func getIAMRoleURL(endpoint string) (*url.URL, error) {
-	if endpoint == "" {
-		endpoint = defaultIAMRoleEndpoint
-	}
-
```
Comment (harshavardhana) [inline pkg/credentials/iam_aws.go:187]: why is this removed?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: pkg/credentials/iam_aws.go [hunk 2]
```diff
@@ -281,6 +277,10 @@ func fetchIMDSToken(client *http.Client, endpoint string) (string, error) {
 // If the credentials cannot be found, or there is an error
 // reading the response an error will be returned.
 func getCredentials(client *http.Client, endpoint string) (ec2RoleCredRespBody, error) {
+	if endpoint == "" {
+		endpoint = defaultIAMRoleEndpoint
+	}
```
Comment (harshavardhana) [inline pkg/credentials/iam_aws.go:282]: can you share the document where it says defaultIAMRoleEndpoint is the default for IMDS token?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: pkg/credentials/iam_aws.go (hunk 1 above)
Comment (iler-dev) [reply pkg/credentials/iam_aws.go:187]: It seemed most appropriate to hoist the block up to `getCredentials()`. I can keep it, but it seems like dead code given it appears that only `getCredentials()` calls `getIAMRoleURL()`.

File: pkg/credentials/iam_aws.go (hunk 2 above)
Comment (iler-dev) [reply pkg/credentials/iam_aws.go:282]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

File: pkg/credentials/iam_aws.go (hunk 1 above)
Comment (harshavardhana) [reply pkg/credentials/iam_aws.go:187]: okay

File: pkg/credentials/iam_aws.go (hunk 2 above)
Comment (harshavardhana) [reply pkg/credentials/iam_aws.go:282]: ah I see its same endpoint.. for newer feature as well.

**Review by harshavardhana — APPROVED** [review]
